monitoring/oplog-populator/dashboard.py

```diff
@@ -0,0 +1,362 @@
+    )],
+    thresholds=[
+        Threshold('red', 0, 0.),
+        Threshold('green', 1, 1.),
+    ])
+
+total_number_buckets = Stat(
+    title='Monitored Buckets',
+    description='Number of buckets configured on the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='-',
+    reduceCalc='last',
+    targets=[Target(
+        expr='sum(' + Metrics.BUCKETS.raw() + ')',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+max_pipeline_size = Stat(
+    title='Max Pipeline Size',
+    description='Maximum size of the mongo pipeline',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='-',
+    reduceCalc='last',
+    format=UNITS.BYTES,
+    orientation='horizontal',
+    targets=[Target(
+        expr='max (' + Metrics.PIPELINE_SIZE.raw() + ')',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+bandwidth = Stat(
+    title='Bandwidth',
+    description='Bandwidth used for reconfiguring the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='-',
+    reduceCalc='last',
+    format=UNITS.BYTES_SEC,
+    orientation='horizontal',
+    targets=[Target(
+        expr='sum (rate(' + Metrics.REQUEST_SIZE() + '))',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+config_applied_success = TimeSeries(
+    title='Connector Reconfiguration Rate',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    legendDisplayMode='hidden',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum (rate(' + Metrics.CONNECTOR_CONFIG_APPLIED('success="true"') + '))',
+    )])
+
+config_applied_failure = TimeSeries(
+    title='Failed Connector Reconfiguration Rate',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    legendDisplayMode='hidden',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum (rate(' + Metrics.CONNECTOR_CONFIG_APPLIED('success="false"') + '))',
+    )])
+
+
+bucket_distribution = PieChart(
+    title='Bucket Distribution',
+    description='How buckets are distributed among the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    legendValues=['percent', 'value'],
+    pieType='donut',
+    legendDisplayMode='table',
+    legendPlacement='right',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_per_connector = TimeSeries(
+    title='Buckets Per Connector',
+    description='Evolution of buckets per connector',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
```

Comment:
For my understanding: what is an "expected" number (order of magnitude) of connectors ?
(what I am thinking is that this graph may become hard to read if there are 10s of connectors)

Reply:
We don't know yet actually, the number will depend on the performance of the connectors, meaning that if a single connector can handle tens of thousands of buckets at a time we'll probably have few connectors, and it's true the other way around.

There's a ticket for doing that performance test which will also help determine how we manage the connectors, for now the number of configured connectors is a set value that can be configured at Zenko CR level. This might change later on to make it dynamic in case of sudden degradation of performance with the addition of buckets to connectors.

Reply:
Thanks for the explanation !